### PR TITLE
Move metadata from setup.cfg --> pyproject.toml

### LINKED ISSUE #303: Move all metadata from setup.cfg --> pyproject.toml
Following Pep 619 ([read more about declations here](https://packaging.python.org/en/latest/specifications/declaring-project-metadata/) we can store all metadata (including deps) in the pyproject.toml file. 

this is the modern best practice for packaging. I am going to make this shift in our project to further modernize it!

### PR DESCRIPTION
**Pull request by lwasser**
closes #303 

## Description

Just following specifications in [PEP 621](https://peps.python.org/pep-0621/) and getting rid of that setup.cfg file. we MAY need to add it back for linting but i may be able to work around it too. 

## Type of change

Select the statement best describes this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [x] Other (please describe)

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [ ] Yes
- [ ] No, i'd like some help with tests
- [x] This change doesn't require tests

## Do all of the CI tests run successfully on this pr?

- [x] Check this when your PR has a green check. 

NOTE: right now the maintainer list is [NOT rendering corrrectly on test pypi.](https://test.pypi.org/)  but everything builds properly locally AFAIK. 

SO.. i messaged the PIP / PYPI maintainer to see if that is a bug or a syntax issue. i did find the same issue with the maintainer list for sourmesh on PyPI. so it could be a bug potentially? not sure.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -10,5 +10,51 @@ version_scheme =  "post-release"
...
+[project]
+name = "stravalib"
+description = "A Python package that makes it easy to access and download data from the Strava V3 REST API."
+keywords = ["strava", "running", "cycling", "athletes"]
+readme = "README.md"
+dynamic =["version"]
+license = {text = "Apache 2.0 License"} 
+authors = [{name="Hans Lellelid", email="hans@xmpl.org"}]
+maintainers = [
+     {name = "Leah Wasser", email = "leah@pyopensci.org"},
```
Comment (lwasser) [inline pyproject.toml:22]: If you look at the maintainers on test pypi you'll see if you hover over the names you get a weird mailto: string and also 2 names are missing (they are what appear in the hover) super weird. but sourmesh has the same issue.

File: pyproject.toml [hunk 2]
```diff
@@ -10,5 +10,51 @@ version_scheme =  "post-release"
...
+     {name = "Hans Lellelid", email = "hans@xmpl.org"},
+     {name = "Jonatan Smoocha"},
```
Comment (lwasser) [inline pyproject.toml:24]: Email is optional but if y'all want your email here or a different version of your name please feel free to add! @yihong0618 @jsamoocha

Comment (lwasser) [reply pyproject.toml:24]: ok there is indeed a pypi bug. confirmed. For this PR i'll remove emails. we can add them back when we have a working fix!

File: pyproject.toml [hunk 3]
```diff
@@ -10,5 +10,51 @@ version_scheme =  "post-release"
...
+     {name = "Yihong"},
```
Comment (lwasser) [inline pyproject.toml:25]:
```suggestion
     {name = "Hans Lellelid"},
     {name = "Jonatan Smoocha"},
     {name = "Yihong"},
```

File: pyproject.toml (hunk 1 above)
Comment (lwasser) [inline pyproject.toml:22]:
```suggestion
     {name = "Leah Wasser"},
```

**Comment by lwasser** [comment]
this is ready to review @jsamoocha @yihong0618 !!

**Comment by lwasser** [comment]
the issue related to PyPI not rendering maintainers w emails correctly is here - https://github.com/pypi/warehouse/issues/9400 just for reference. oh and docs are failing - let me check ...

**Comment by lwasser** [comment]
ok breaking because of an instagram link? it's completely unrelated to this PR. it's a rate-limiter error

File: pyproject.toml [hunk 4]
```diff
@@ -10,5 +10,51 @@ version_scheme =  "post-release"
...
+description = "A Python package that makes it easy to access and download data from the Strava V3 REST API."
+keywords = ["strava", "running", "cycling", "athletes"]
+readme = "README.md"
+dynamic =["version"]
+license = {text = "Apache 2.0 License"} 
+authors = [{name="Hans Lellelid", email="hans@xmpl.org"}]
+maintainers = [
+     {name = "Leah Wasser"},
+     {name = "Hans Lellelid"},
+     {name = "Jonatan Smoocha"},
```
Comment (jsamoocha) [inline pyproject.toml:24]:
```suggestion
     {name = "Jonatan Samoocha"},
```

**Review by jsamoocha — APPROVED** [review]
Besides that one typo, everything looks good to merge!

**Comment by lwasser** [comment]
thanks!! sorry about that typo in your name. goodness.

**Comment by lwasser** [comment]
darnit meant to rebase and merge :(

**Pull request merged**